app/services/xapi/completed_target_service.rb

```diff
@@ -0,0 +1,29 @@
+module Xapi
+  class CompletedTargetService < Xapi::BaseService
+    def initialize(agent, target)
+      @agent = agent
+      @object = object(target)
+    end
+
+    def set_statement
+      @statement = Xapi.create_statement(actor: @agent, verb: verb, object: @object)
+    end
+
+    private
+
+    def verb
+      Xapi.create_verb(id: 'https://w3id.org/xapi/dod-isd/verbs/completed-assignment', name: 'completed assignment')
+    end
+
+    def object(target)
+      params = {
+          id: url_helpers.target_url(target, host: target.course.school.domains.primary.fqdn),
```

Comment:
Is this a good choice of id?
What happens if 2 different users complete the same target for the same school?
Maybe create an id by hashing that route  with `@agents` email and a timestamp?

Reply:
Nvm. But wouldn't this id be more of object type or tag, no?